src/cpu/core_dynrec/decoder_basic.h

```diff
@@ -221,39 +221,44 @@ static void decode_advancepage(void) {
 }
 
 // fetch the next byte of the instruction stream
-static Bit8u decode_fetchb(void) {
-	if (GCC_UNLIKELY(decode.page.index>=4096)) {
+static uint8_t decode_fetchb()
+{
+	if (GCC_UNLIKELY(decode.page.index >= 4096)) {
 		decode_advancepage();
 	}
-	decode.page.wmap[decode.page.index]+=0x01;
+	decode.page.wmap[decode.page.index] += 0x01;
 	decode.page.index++;
-	decode.code+=1;
-	return mem_readb(decode.code-1);
+	decode.code += 1;
+	return mem_readb(decode.code - 1);
 }
 // fetch the next word of the instruction stream
-static Bit16u decode_fetchw(void) {
-	if (GCC_UNLIKELY(decode.page.index>=4095)) {
-   		Bit16u val=decode_fetchb();
-		val|=decode_fetchb() << 8;
+static uint16_t decode_fetchw()
+{
+	if (GCC_UNLIKELY(decode.page.index >= 4095)) {
+		Bit16u val = decode_fetchb();
+		val |= decode_fetchb() << 8;
 		return val;
 	}
-	*(Bit16u *)&decode.page.wmap[decode.page.index]+=0x0101;
```

Comment:
PPC: Fix needs to be aligned and native endian (not swapped to little-endian like the host_* functions do).